components/engine_rocks/src/write_batch.rs

```diff
@@ -7,134 +7,211 @@ use rocksdb::{Writable, WriteBatch as RawWriteBatch, DB};
 
 use crate::{engine::RocksEngine, options::RocksWriteOptions, util::get_cf_handle};
 
+const WRITE_BATCH_MAX_BATCH: usize = 16;
+const WRITE_BATCH_LIMIT: usize = 16;
+
 impl WriteBatchExt for RocksEngine {
-    type WriteBatch = RocksWriteBatch;
+    type WriteBatch = RocksWriteBatchVec;
 
     const WRITE_BATCH_MAX_KEYS: usize = 256;
 
-    fn write_batch(&self) -> RocksWriteBatch {
-        RocksWriteBatch::new(self.as_inner().clone())
+    fn write_batch(&self) -> RocksWriteBatchVec {
+        RocksWriteBatchVec::new(Arc::clone(self.as_inner()), WRITE_BATCH_LIMIT, 1)
     }
 
-    fn write_batch_with_cap(&self, cap: usize) -> RocksWriteBatch {
-        RocksWriteBatch::with_capacity(self, cap)
+    fn write_batch_with_cap(&self, cap: usize) -> RocksWriteBatchVec {
+        RocksWriteBatchVec::with_capacity(self, cap)
     }
 }
 
-pub struct RocksWriteBatch {
+/// `RocksWriteBatchVec` is for method `MultiBatchWrite` of RocksDB, which splits a large WriteBatch
+/// into many smaller ones and then any thread could help to deal with these small WriteBatch when it
+/// is calling `MultiBatchCommit` and wait the front writer to finish writing. `MultiBatchWrite` will
+/// perform much better than traditional `pipelined_write` when TiKV writes very large data into RocksDB.
+/// We will remove this feature when `unordered_write` of RocksDB becomes more stable and becomes compatible
+/// with Titan.
+pub struct RocksWriteBatchVec {
     db: Arc<DB>,
-    wb: RawWriteBatch,
+    wbs: Vec<RawWriteBatch>,
+    save_points: Vec<usize>,
+    index: usize,
+    cur_batch_size: usize,
+    batch_size_limit: usize,
+    support_write_batch_vec: bool,
 }
 
-impl RocksWriteBatch {
-    pub fn new(db: Arc<DB>) -> RocksWriteBatch {
-        let wb = RawWriteBatch::new();
-        RocksWriteBatch { db, wb }
-    }
-
-    pub fn with_capacity(engine: &RocksEngine, cap: usize) -> RocksWriteBatch {
+impl RocksWriteBatchVec {
```

Comment:
Update tests under engine_traits_tests/../write_batch.rs to test both versions (especially save point).